src/components/SearchForm/SearchForm.js

```diff
@@ -1,22 +1,46 @@
 import './SearchForm.css';
-
+import * as validate from '../../utils/validateForm';
+import useForm from '../../utils/useForm';
+import { useEffect } from 'react';
 function SearchForm({ onSearchSubmit }) {
-  function handleSubmit(e) {
-    e.preventDefault();
-    onSearchSubmit();
-  }
+  const { handleChange, onSubmit, values, errors } = useForm(
+    validate.validateSearch
+  );
+  const handleSubmit = (e) => {
+    onSubmit(e);
+    if (values.search) {
+      onSearchSubmit(values.search);
+    }
+  };
+
+  useEffect(() => {
+    values.search = '';
+    // eslint-disable-next-line
+  }, []);
+
   return (
     <form className='search-form' onSubmit={handleSubmit}>
-      <input
-        className='search-form__input'
-        type='search'
-        id='search'
-        name='search'
-        placeholder='Enter topic'
-      ></input>
-      <button className='search-form__button' onClick={handleSubmit}>
-        Search
-      </button>
+      <div className='search-form__wrapper'>
+        <input
+          className='search-form__input'
+          type='search'
+          id='search'
+          name='search'
+          placeholder='Enter topic'
+          onChange={handleChange}
+          value={values.search}
+        ></input>
+        <button className='search-form__button' onClick={handleSubmit}>
```

Comment:
`onClick={handleSubmit}`

please, delete it


Submit handlers are added only to tags `form` with event `submit` rather than to submit buttons with event `click` because it also automatically handles `Enter` presses